src/services/targets.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Service } from 'typedi';
+import { getRepository } from 'typeorm';
+import { Target } from '@entities/target.entity';
+import { TargetNotSavedException } from '@exception/targets/target-not-saved.exception';
+import { TargetErrorsMessages } from '@constants/errorMessages';
+
+@Service()
+export class TargetsService {
+  private readonly targetRepository = getRepository<Target>(Target);
+
+  async canCreateTargets(userId: number): Promise<boolean> {
+    try {
+      const numberTargets = await this.targetRepository.count({ where: { userId } });
+      return numberTargets < 10;
+    } catch (error) {
+      throw new Error(`${error}`);
+    }
```

Comment:
This is the right way of doing things if you wanna catch error on async functions